src/ADC.c

```diff
@@ -1,9 +1,17 @@
 #include "stm32f4xx.h"
+#include "FreeRTOSConfig.h"
+#include "FreeRTOS.h"
+#include "task.h"
 #include "ADC.h"
+#include "stdlib.h"
 
-uint16_t ADC_Values[17][5];
+#define sensor_quantity 17
```

Comment:
You shouldn't have two defines with the same name. You have sensor_quantity and amount_of_recored_values defined in the header file as well. The compiler usually will give you a warning for this. I would recommend putting it in the header file since if you want other files to access ADC_Values, you probably want it to have access to these define.